src/main/java/com/manywho/services/dummy/account/repositories/AccountRepository.java

```diff
@@ -18,44 +20,104 @@ public AccountRepository(Sql2o sql2o) {
+
+            if (columnname.equals("id")) {
+
+                if (criteriaType == CriteriaType.GreaterThan) {
+                    sql = "SELECT * FROM account WHERE id > " + value;
+                } else if (criteriaType == CriteriaType.GreaterThanOrEqual) {
+                    sql = "SELECT * FROM account WHERE id >= " + value;
+                } else if (criteriaType == CriteriaType.LessThan) {
+                    sql = "SELECT * FROM account WHERE id < " + value;
+                } else if (criteriaType == CriteriaType.LessThanOrEqual) {
+                    sql = "SELECT * FROM account WHERE id <= " + value;
+                } else if (criteriaType == CriteriaType.Equal) {
+                    sql = "SELECT * FROM account WHERE id == " + value;
+                } else if (criteriaType == CriteriaType.NotEqual) {
+                    sql = "SELECT * FROM account WHERE id != " + value;
+                } else if (criteriaType == CriteriaType.IsEmpty) {
+                    sql = (Boolean.valueOf(filter.getWhere().get(0).getContentValue())) ?
+                            "SELECT * FROM account WHERE id is null" :
+                            "SELECT * FROM account WHERE id is not null";
+                } else {
+                    throw new ServiceProblemException(400, "Integer type \"id\" only supports EQUAL, NOT_EQUAL, GREATER_THAN, GREATER_THAN_OR_EQUAL, LESS_THAN, LESS_THAN_OR_EQUAL and IS_EMPTY operations");
+                }
+            } else if (columnname.equals("company") || columnname.equals("name") || columnname.equals("stateid")) {
+
+                if (criteriaType == CriteriaType.StartsWith) {
+                    sql = "SELECT * FROM account WHERE " + columnname + " LIKE \'" + value + "%\'";
+                } else if (criteriaType == CriteriaType.EndsWith) {
+                    sql = "SELECT * FROM account WHERE " + columnname + " LIKE \'%" + value + "\'";
+                } else if (criteriaType == CriteriaType.IsEmpty) {
+                    sql = (Boolean.valueOf(filter.getWhere().get(0).getContentValue())) ?
+                            "SELECT * FROM account WHERE " + columnname + " is null" :
+                            "SELECT * FROM account WHERE " + columnname + " is not null";
+                } else if (criteriaType == CriteriaType.Contains) {
+                    sql = "SELECT * FROM account WHERE " + columnname + " LIKE \'%" + value + "%\'";
+                } else if (criteriaType == CriteriaType.Equal) {
+                    sql = "SELECT * FROM account WHERE " + columnname + " == \'" + value + "\'";
+                } else if (criteriaType == CriteriaType.NotEqual) {
+                    sql = "SELECT * FROM account WHERE " + columnname + " != \'" + value + "\'";
+                } else {
+                    throw new ServiceProblemException(400, "String CriteriaType only supports EQUAL, NOT_EQUAL, STARTS_WITH, ENDS_WITH, CONTAINS and IS_EMPTY operations.");
+                }
+            }
```

Comment:
This part of the code is not really reusable, I think it would be better if you create the query in one method and validate the criteria in other. Also you can use a switch that is more easy to read.

Reply:
I've extracted this block code into two methods , one for integer, one for string type, is this the right way to do it?
Also change from if to switch to make it easier to read.

Reply:
yes that will work too.